test/e2e/network_policy.go

```diff
@@ -0,0 +1,509 @@
+		By("Creating client-a which will be able to contact the server since isolation is off.")
+		podClientA := createClientPod(f, ns, clientAName, service, listeningPort1)
+		defer func() {
+			By("Cleaning up the pod client-a")
+			if err := f.Client.Pods(ns.Name).Delete(podClientA.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podClientA.Name, err)
+			}
+		}()
+
+		framework.Logf("Waiting for client-a to to come up.")
+		err = framework.WaitForPodRunningInNamespace(f.Client, podClientA)
+		Expect(err).NotTo(HaveOccurred(), "waiting for Client-A to run")
+		framework.Logf("Waiting for client-a to complete.")
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientA.Name, ns.Name)
+		Expect(err).NotTo(HaveOccurred(), "checking client-a could communicate with server.")
+
+		framework.Logf("Enabling network isolation.")
+		setNamespaceIsolation(f, ns, "DefaultDeny")
+
+		// Create a pod with name 'client-b', which will attempt to comunicate with the server,
+		//  but should not be able to now that isolation is on.
+		By("Creating client-b which should *not* be able to contact the server.")
+		podClientB := createClientPod(f, ns, clientBName, service, listeningPort1)
+		defer func() {
+			By("Cleaning up the pod client-b.")
+			if err := f.Client.Pods(ns.Name).Delete(podClientB.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podClientB.Name, err)
+			}
+		}()
+
+		framework.Logf("Waiting for client-b to run.")
+		err = framework.WaitForPodRunningInNamespace(f.Client, podClientB)
+		Expect(err).NotTo(HaveOccurred(), "waiting for Client-B to run")
+
+		framework.Logf("Waiting for client-b to complete.")
+		err = framework.WaitForPodSuccessInNamespace(f.Client, podClientB.Name, ns.Name)
+		Expect(err).To(HaveOccurred(), "checking client-b could not communicate with server")
+	})
+
+	It("should enforce policy based on PodSelector [Feature:NetworkPolicy]", func() {
+		ns := f.Namespace
+		setNamespaceIsolation(f, ns, "DefaultDeny")
+
+		By("Creating a simple server.")
+		podServer, service := createServerPod(f, ns, serverName, []int{listeningPort1})
+		defer func() {
+			By("Cleaning up the server.")
+			if err := f.Client.Pods(ns.Name).Delete(podServer.Name, nil); err != nil {
+				framework.Failf("unable to delete pod %v: %v", podServer.Name, err)
+			}
+		}()
+		defer func() {
+			By("Cleaning up the server's service.")
+			if err := f.Client.Services(ns.Name).Delete(service.Name); err != nil {
+				framework.Failf("unable to delete svc %v: %v", service.Name, err)
+			}
+		}()
+		framework.Logf("Waiting for Server to come up.")
+		err := framework.WaitForPodRunningInNamespace(f.Client, podServer)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating a network policy for the server which allows traffic from the pod 'client-a'.")
+		policy, err := f.Client.NetworkPolicies(ns.Name).Create(&extensions.NetworkPolicy{
+			ObjectMeta: api.ObjectMeta{
+				Name: "test-policy",
```

Comment:
A more descriptive name here would be nice - something like `Name: "allow-client-a"` maybe?